plugin/documents.py

```diff
@@ -241,15 +236,12 @@ def on_session_shutdown_async(self, session: Session) -> None:
             session.config.erase_view_status(self.view)
 
     def _diagnostics_async(
-        self, throw_if_stale: bool = False
+        self, allow_stale: bool = False
```

Comment:
@rchl do you see a way to remove the `allow_stale`, to not even exist?
Would that be problematic or do you see it as necessary for this to work?

Reply:
Not sure if in this form or different but it's necessary. We need to receive diagnostics in latest state and check whether there are relevant diagnostics. Otherwise we'll receive empty list of diagnostics if those are stale and make wrong decision based on that.

Reply:
There was a comment `# do not provide stale diagnostics` introduced in b84f25ec917196d860032bea0d5cb20942f13e54
made by @rwols . 

What bad can happen if we provide stale diagnostics? 
1. If nothing bad can happen, I would remove the `allow_stale` arg.

2. The first bad thing that comes to my mind, If we for example allow stale diagnostics, code actions on save would be affected, and I guess something bad could happen there, but I am not sure. 

But still does anyone thing that we can have something like this?
```diff
    def _diagnostics_async(
        self, allow_stale: bool = False
    ) -> Generator[Tuple[SessionBufferProtocol, List[Tuple[Diagnostic, sublime.Region]]], None, None]:
        change_count = self.view.change_count()
        for sb in self.session_buffers_async():
-            # do not provide stale diagnostics
-            if sb.diagnostics_version == change_count or allow_stale:
-                yield sb, sb.diagnostics
+            yield sb, sb.diagnostics
```

Reply:
I think your point about code actions on save is a good argument already for keeping it.